File: python/83.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if not head or not head.next:
        	return head
        cur = head
        while cur:
        	while cur.next and cur.val == cur.next.val:
        		cur.next = cur.next.next
        	cur = cur.next
        return head

#Runtime: 68 ms
#Your runtime beats 17.98 % of python3 submissions.

File: JAVA/83.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
    	if (head == null || head.next == null) {
    		return head;
    	}
    	ListNode cur = head;
    	while (cur != null) {
    		while (cur.next !=null && cur.val == cur.next.val) {
    			cur.next = cur.next.next;
    		}
    		cur = cur.next;
    	}
        return head;
    }
}

#Runtime: 1 ms
#Your runtime beats 26.97 % of java submissions.
#space: O(1)
#time: O(n)